pkg/controller/crdoffloadstore/active.go

```diff
@@ -0,0 +1,75 @@
+package crdoffloadstore
+
+import (
+	"context"
+	"time"
+
+	"github.com/flyteorg/flytestdlib/promutils"
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/flyteorg/flytepropeller/pkg/apis/flyteworkflow/v1alpha1"
+)
+
+type activeCRDOffloadMetrics struct {
+	TotalItems   prometheus.Gauge
+	CacheHit     prometheus.Counter
+	CacheMiss    prometheus.Counter
+	ReadError    prometheus.Counter
+	FetchLatency promutils.StopWatch
+}
+
+type activeCRDOffloadStore struct {
+	crdOffloadStore CRDOffloadStore
+	metrics         *activeCRDOffloadMetrics
+	store           map[string]*v1alpha1.StaticWorkflowData
+}
+
+func (a *activeCRDOffloadStore) Get(ctx context.Context, dataReference v1alpha1.DataReference) (*v1alpha1.StaticWorkflowData, error) {
+	location := dataReference.String()
+	if m, ok := a.store[location]; ok {
+		a.metrics.CacheHit.Inc()
+		return m, nil
+	}
+
+	a.metrics.CacheMiss.Inc()
+
+	timer := a.metrics.FetchLatency.Start()
+	staticWorkflowData, err := a.crdOffloadStore.Get(ctx, dataReference)
+	timer.Stop()
+	if err != nil {
+		a.metrics.ReadError.Inc()
+		return nil, err
+	}
+
+	a.metrics.TotalItems.Inc()
+	a.store[location] = staticWorkflowData
+	return staticWorkflowData, nil
+}
+
+func (a *activeCRDOffloadStore) Remove(ctx context.Context, dataReference v1alpha1.DataReference) error {
+	if err := a.crdOffloadStore.Remove(ctx, dataReference); err != nil {
+		return err
+	}
+
+	a.metrics.TotalItems.Dec()
+	delete(a.store, dataReference.String())
+
+	return nil
+}
+
+func NewActiveCRDOffloadStore(crdOffloadStore CRDOffloadStore, scope promutils.Scope) CRDOffloadStore {
```

Comment:
this is a rename, right?

Reply:
Correct. I think that the "Inmemory" in terms of a CRD offload store implies that there is no backing store. Technically the LRU cache is also "in memory". So maybe "active" means that we are caching all of the workflows that are currently active. I'm not tied to the naming, OK changing if you feel stronlgy.